internal/services/dispatch/dispatchService_integration_test.go

```diff
@@ -0,0 +1,179 @@
+	"subscription-api/pkg/db"
+	"subscription-api/pkg/utils"
+	"testing"
+
+	currency_service "subscription-api/internal/services/currency"
+	currency_grpc "subscription-api/internal/services/currency/grpc"
+
+	grpc_client "subscription-api/pkg/grpc"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type DispatchServiceSuite struct {
+	suite.Suite
+	cs interface {
+		Stop()
+	}
+	service     *dispatchService
+	l           config.Logger
+	dbContainer *testhelpers.PostgresContainer
+	ctx         context.Context
+	mailman     *stubs.MailmanStub
+}
+
+const CurrencyServiceURL = "localhost:4040"
+
+func (s *DispatchServiceSuite) startCurrencyServiceServer() {
+	lis, err := net.Listen("tcp", CurrencyServiceURL)
+	s.NoErrorf(err, "failed to listen url")
+
+	server := grpc.NewServer()
+	currency_grpc.RegisterCurrencyServiceServer(server,
+		currency_service.NewCurrencyServiceServer(
+			currency_service.NewCurrencyService(stubs.NewCurrencyAPIClient()),
+			s.l,
+		))
+	s.cs = server
+
+	go func() {
+		if err := server.Serve(lis); err != nil {
+			s.Fail(err.Error())
+		}
+	}()
+}
+
+func (s *DispatchServiceSuite) SetupSuite() {
+	s.l = config.InitLogger(config.TestMode)
+	s.startCurrencyServiceServer()
+
+	s.ctx = context.Background()
+	cont, err := testhelpers.CreatePostgresContainer(s.ctx, testhelpers.DBParams{
+		Database: "subs",
+		Username: "postgres",
+		Password: "pass",
+	})
+	if err != nil {
+		s.Failf("failed to create postgres container: %v", err.Error())
+	}
+	s.dbContainer = cont
+
+	currencyServiceConn, err := grpc.NewClient(
+		CurrencyServiceURL,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	s.NoErrorf(err, "failed to create connection to currency service")
+
+	mailman := stubs.NewMailmanStub()
+	s.mailman = mailman
+
+	s.service = NewDispatchService(&DispatchServiceParams{
+		Logger: s.l,
+		Store: store.NewStore(
+			utils.Must(db.NewPostrgreSQL(
+				s.dbContainer.ConnectionString,
+				sql.PostgeSQLMigrationsUp(nil),
+			)),
+			db.IsPqError,
+		),
+		Mailman:         mailman,
+		CurrencyService: grpc_client.NewCurrencyServiceClient(currencyServiceConn),
+	})
+}
+
+func (s *DispatchServiceSuite) TearDownSuite() {
+	s.cs.Stop()
+	if err := s.dbContainer.Terminate(s.ctx); err != nil {
+		log.Fatalf("error terminating database container: %s", err)
+	}
+}
+
+func (s *DispatchServiceSuite) SetupTest() {}
+
+func (s *DispatchServiceSuite) TestIntegration_GetAllDispatches() {
+	ctx := context.Background()
+
+	dispatches1, err1 := s.service.GetAllDispatches(ctx)
+	dispatches2, err2 := s.service.GetAllDispatches(ctx)
+
+	s.NoError(err1)
+	s.NoError(err2)
+	s.Equal(1, len(dispatches1))
+	s.Equal(1, len(dispatches2))
+	s.Equal(dispatches1, dispatches2)
+	s.Equal(dispatches1[0].Id, services.USD_UAH_DISPATCH_ID)
+}
+
+func (s *DispatchServiceSuite) TestIntegration_SendDispatch() {
```

Comment:
I see problem in this test, that you calls another methods in service. Try fixture approach where you have predifined rows in DB, so you will only need to test SendDispatch.

Reply:
Updated test flow to use only tested method of service